crates/dids/src/method/key.rs

```diff
@@ -0,0 +1,124 @@
+use crate::did::Did;
+use crate::method::{DidMethod, DidMethodError, DidResolutionResult};
+use async_trait::async_trait;
+use crypto::key::{Key, KeyType};
+use crypto::key_manager::KeyManager;
+use did_method_key::DIDKey as SpruceDidKeyMethod;
+use ssi_dids::did_resolve::{DIDResolver, ResolutionInputMetadata};
+use ssi_dids::{DIDMethod, Source};
+use std::sync::Arc;
+
+/// Concrete implementation for a did:key DID
+pub struct DidKey {
+    uri: String,
+    key_manager: Arc<dyn KeyManager>,
+}
+
+pub struct DidKeyCreateOptions {
+    pub key_type: KeyType,
+}
+
+impl Did for DidKey {
+    fn uri(&self) -> &str {
+        &self.uri
+    }
+
+    fn key_manager(&self) -> &Arc<dyn KeyManager> {
+        &self.key_manager
+    }
+}
+
+#[async_trait]
+impl DidMethod<DidKey, DidKeyCreateOptions> for DidKey {
+    const NAME: &'static str = "key";
+
+    fn create(
+        key_manager: Arc<dyn KeyManager>,
+        options: DidKeyCreateOptions,
+    ) -> Result<DidKey, DidMethodError> {
+        let key_alias = key_manager.generate_private_key(options.key_type)?;
+        let public_key =
+            key_manager
+                .get_public_key(&key_alias)?
+                .ok_or(DidMethodError::DidCreationFailure(
+                    "PublicKey not found".to_string(),
+                ))?;
+
+        let uri = SpruceDidKeyMethod
+            .generate(&Source::Key(public_key.jwk()))
+            .ok_or(DidMethodError::DidCreationFailure(
+                "Failed to generate did:key".to_string(),
+            ))?;
+
+        Ok(DidKey { uri, key_manager })
+    }
+
+    async fn resolve_uri(did_uri: &str) -> DidResolutionResult {
+        let input_metadata = ResolutionInputMetadata::default();
+        let (did_resolution_metadata, did_document, did_document_metadata) =
+            SpruceDidKeyMethod.resolve(did_uri, &input_metadata).await;
+
+        DidResolutionResult {
+            did_resolution_metadata,
+            did_document,
+            did_document_metadata,
+            ..Default::default()
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crypto::key_manager::local_key_manager::LocalKeyManager;
+    use ssi_dids::did_resolve::ERROR_INVALID_DID;
+
+    fn create_did_key(key_type: KeyType) -> DidKey {
+        let key_manager = Arc::new(LocalKeyManager::new_in_memory());
+
+        DidKey::create(key_manager, DidKeyCreateOptions { key_type })
+            .expect("DidKey Ed25519 creation failed")
```

Comment:
Error message is hardcoded to `Ed25519` as the key type rather than using the `key_type` parameter

Reply:
This actually begs the question... should we define default key types?

Reply:
Fixed it as I rebased to main!

Rust doesn't have the concept of "default parameter values" like a lot of other languages have. We _could_ define a default value for ANY `KeyType` if it's not present, but that didn't seem right to me. What may be a good default for one did method may not be a good default for another.

The other approach we could take is the builder pattern, which would allow us to define default values if the builder doesn't have a value set for a required field.

Reply:
Another options is we could conform `DidKeyCreateOptions` (and the others like `DidJwkCreateOptions`) and have it conform to the `Default` trait ([documentation](https://doc.rust-lang.org/std/default/trait.Default.html)).

Then we could define whatever default options we think are good for creating `did:key` with.

The only downside, the user still needs to provide a value when calling this function. It would just change to the following:
`DidKey::create(key_manager, DidKeyCreateOptions::default())`

Reply:
Just briefly perusing `web5-js` and `web5-kt`, we're offering at least some amount of default-value DX. I think this warrants an issue on the backlog, if only for the sake of discussion to rule it out. I don't love simplicity insofar as it robs the developer of valuable knowledge, but intuitively, I think it would be wise to offer a simple `create()` DX where the developer can easily get-off-zero. Which further begs the question of if the key manager should also have a default value.

Reply:
A compelling counter argument, against supporting default values, is that Rust is not a junior dev friendly language. Which is to say, not a lot of newbie developers are using Rust. In which case, default values only serve to add unnecessary overhead and confusion. 

Idk just trying to steel man the counter case, I still think intuitively, default values are a good DX.

Reply:
Late in the day for me so I'm rambling, but anyways this isn't a priority whatsoever IMO

Reply:
https://github.com/TBD54566975/web5-rs/issues/45